cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -40,11 +40,12 @@ type MasterConfiguration struct {
 	KubernetesVersion string
 	CloudProvider     string
 	AuthorizationMode string
+	CertAltNames      []string
```

Comment:
x509 certificate Subject Alternate Name extensions are actually typed.  You can include DNS names, URIs and IP addresses.  I see below that you auto-detect if it is a DNS name or IP address. 

We probably should reflect that typing explicitly.  Either with separate lists for IPs and DNS or by having some sort of prefix here: `dns:foo.com`, `ip:x.x.x.x`.  I'd vote for separate parameters.  `CertExtraDNS` and `CertExtraIP`.

We should then document that these are added to the "Subject Alternate Name" section of the certificate.

But -- we are down to the wire so I won't insist.  But if we want to do more than DNS or IP addresses we'll have to add more parameters and be more explicit.